src/lib.rs

```diff
@@ -322,6 +318,19 @@ impl<T: Translation> Mapping<T> {
                         (dd.flags(), dd.output_address())
                     };
 
+                    if !mr.is_block(level) {
+                        // The region being mapped is smaller than a block mapping at the current
+                        // level. Given that the block mapping is live, replacing it with a table
+                        // mapping is not allowed by BBM. However, if the output address and flags
+                        // of the new smaller mapping match the ones of the block mapping, nothing
+                        // needs to be done and so it can be allowed.
+                        if flags != d.flags()
+                            || align_down(oa.0, granularity_at_level(level)) != d.output_address().0
```

Comment:
The same issue applies here, this needs to check the offset from the block physical address, not align down the page physical address.